File: resellos/capture_ebay_sold.js
#!/usr/bin/env node
/**
 * eBay Sold Proof Screenshot Capture
 * Captures eBay sold/finished listings for toilet tank flush lever products
 * Saves screenshots for manual review
 */

const { chromium } = require('playwright');
const path = require('path');
const fs = require('fs');

const OUT_DIR = path.join(__dirname, 'ebay-sold-proofs');
if (!fs.existsSync(OUT_DIR)) fs.mkdirSync(OUT_DIR, { recursive: true });

const SEARCHES = [
  'toilet+tank+flush+lever+replacement',
  'toilet+tank+handle+replacement',
  'universal+toilet+tank+lever',
  'brass+toilet+tank+lever',
  'Korky+toilet+tank+lever',
  'Fluidmaster+toilet+tank+lever',
  'Danco+toilet+tank+lever',
  'toilet+tank+lever+2+inch',
  'metal+toilet+flush+lever+handle',
];

const BASE_URL = 'https://www.ebay.com/sch/i.html?_nkw=SEARCH&LH_Sold=1&_sop=12';

async function captureSearch(page, searchTerm, index) {
  const url = BASE_URL.replace('SEARCH', searchTerm);
  const filename = `${String(index).padStart(2, '0')}_${searchTerm.replace(/\+/g, '_')}.png`;
  const filepath = path.join(OUT_DIR, filename);

  console.log(`[${index}] Navigating: ${searchTerm}`);
  try {
    await page.goto(url, { waitUntil: 'domcontentloaded', timeout: 15000 });
    await page.waitForTimeout(2000);

    // Check if we got blocked
    const title = await page.title();
    if (title.includes('Access Denied') || title.includes('403')) {
      console.log(`  BLOCKED: ${title}`);
      return null;
    }

    // Scroll to load content
    await page.evaluate(() => window.scrollTo(0, 400));
    await page.waitForTimeout(1000);

    await page.screenshot({ path: filepath, fullPage: false });
    console.log(`  Saved: ${filename} (${title})`);
    return filepath;
  } catch (e) {
    console.log(`  ERROR: ${e.message}`);
    return null;
  }
}

async function captureItem(page, itemUrl, index) {
  const filename = `${String(index).padStart(3, '0')}_item.png`;
  const filepath = path.join(OUT_DIR, filename);

  console.log(`  Item: ${itemUrl}`);
  try {
    await page.goto(itemUrl, { waitUntil: 'domcontentloaded', timeout: 15000 });
    await page.waitForTimeout(2000);

    const title = await page.title();
    if (title.includes('Access Denied') || title.includes('403')) {
      console.log(`    BLOCKED`);
      return null;
    }

    await page.screenshot({ path: filepath, fullPage: false });
    console.log(`    Saved: ${filename}`);
    return filepath;
  } catch (e) {
    console.log(`    ERROR: ${e.message}`);
    return null;
  }
}

async function main() {
  console.log('Starting eBay sold proof capture...');
  console.log(`Output directory: ${OUT_DIR}\n`);

  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext({
    viewport: { width: 1280, height: 900 },
    userAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36',
  });
  const page = await context.newPage();

  let captured = 0;
  const results = [];

  // Capture each search page
  for (let i = 0; i < SEARCHES.length; i++) {
    const result = await captureSearch(page, SEARCHES[i], i + 1);
    if (result) {
      captured++;
      results.push({ search: SEARCHES[i], file: result });
    }
    await page.waitForTimeout(1500);
  }

  // Now try to extract item links from the first search results and capture items
  console.log('\n--- Re-visiting top results for item captures ---');
  const topSearch = SEARCHES[0];
  const topUrl = BASE_URL.replace('SEARCH', topSearch);
  await page.goto(topUrl, { waitUntil: 'domcontentloaded', timeout: 15000 });
  await page.waitForTimeout(2000);

  // Extract item links from search results
  const itemLinks = await page.$$eval('a[href*="/itm/"]', links =>
    links
      .map(a => a.href)
      .filter(h => h.includes('/itm/') && !h.includes('ebay.com/itm/157261456168')) // dedupe
      .slice(0, 8)
  );

  console.log(`Found ${itemLinks.length} item links from top search`);

  for (let j = 0; j < Math.min(itemLinks.length, 6); j++) {
    const result = await captureItem(page, itemLinks[j], 100 + j);
    if (result) {
      captured++;
      results.push({ item: itemLinks[j], file: result });
    }
    await page.waitForTimeout(1500);
  }

  await browser.close();

  // Write results manifest
  const manifest = {
    captured_at: new Date().toISOString(),
    total_captured: captured,
    results,
    searches_run: SEARCHES.length,
  };

  fs.writeFileSync(path.join(OUT_DIR, 'manifest.json'), JSON.stringify(manifest, null, 2));
  console.log(`\nDone. Captured ${captured} screenshots. Manifest written to ${OUT_DIR}/manifest.json`);
}

main().catch(e => {
  console.error('Fatal:', e);
  process.exit(1);
});

File: resellos/capture_ebay_sold.py
#!/usr/bin/env python3
"""
eBay Sold Proof Screenshot Capture
Captures eBay sold/completed listings for toilet tank flush lever products
Saves screenshots for manual review
"""

from playwright.sync_api import sync_playwright
import subprocess, json, os
from pathlib import Path

OUT_DIR = Path("/Users/admin/CascadeProjects/is44ks-shop/resellos/ebay-sold-proofs")
OUT_DIR.mkdir(exist_ok=True)

SEARCHES = [
    ("toilet+tank+flush+lever+replacement", "01_lever"),
    ("toilet+tank+handle+replacement", "02_handle"),
    ("universal+toilet+tank+lever", "03_universal"),
    ("brass+toilet+tank+lever", "04_brass"),
    ("Korky+toilet+tank+lever", "05_korky"),
    ("Fluidmaster+toilet+tank+lever", "06_fluidmaster"),
    ("Danco+toilet+tank+lever", "07_danco"),
    ("toilet+tank+lever+2+inch", "08_2inch"),
    ("metal+toilet+flush+lever+handle", "09_metal"),
]

BASE_URL = "https://www.ebay.com/sch/i.html?_nkw={search}&LH_Sold=1&_sop=12"

results = []

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    context = browser.new_context(
        viewport={"width": 1280, "height": 900},
        user_agent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
    )
    page = context.new_page()

    for search_term, label in SEARCHES:
        url = BASE_URL.format(search=search_term)
        screenshot_path = OUT_DIR / f"search_{label}.png"
        print(f"[{label}] Navigating: {search_term}")

        try:
            page.goto(url, wait_until="domcontentloaded", timeout=15000)
            page.wait_for_timeout(2000)
            title = page.title()

            if "Access Denied" in title or "403" in title:
                print(f"  BLOCKED: {title}")
            else:
                page.evaluate("window.scrollTo(0, 400)")
                page.wait_for_timeout(1000)
                page.screenshot(path=str(screenshot_path), full_page=False)
                print(f"  Saved: {screenshot_path.name}")

                # Extract item links
                item_links = page.evaluate("""() => {
                    const links = Array.from(document.querySelectorAll('a[href*="/itm/"]'));
                    return links
                        .map(a => a.href)
                        .filter(u => u.match(/\\/itm\\/\\d+/))
                        .filter(u => !u.includes('ebay.com/itm/157261456168'));
                }""")
                print(f"  Found {len(item_links)} item links")
                results.append({
                    "search": search_term,
                    "label": label,
                    "screenshot": str(screenshot_path),
                    "title": title,
                    "item_count": len(item_links),
                    "items": item_links[:6],
                })

        except Exception as e:
            print(f"  ERROR: {e}")

        page.wait_for_timeout(1500)

    # Capture individual item pages
    print("\n--- Capturing individual sold items ---")
    item_id = 1
    seen_urls = set()

    for search_result in results:
        for item_url in search_result.get("items", []):
            if item_url in seen_urls:
                continue
            seen_urls.add(item_url)

            screenshot_path = OUT_DIR / f"item_{item_id:03d}.png"
            print(f"  Item {item_id}: {item_url[:80]}")

            try:
                page.goto(item_url, wait_until="domcontentloaded", timeout=15000)
                page.wait_for_timeout(2000)

                title = page.title()
                if "Access Denied" in title or "403" in title:
                    print(f"    BLOCKED")
                    continue

                page.screenshot(path=str(screenshot_path), full_page=False)
                print(f"    Saved: {screenshot_path.name}")

                # Extract sold info
                sold_info = page.evaluate("""() => {
                    const text = document.body.innerText;
                    const soldMatch = text.match(/(\\d+)\\s*sold/i);
                    const priceMatch = text.match(/\\$[\\d,]+\\.\\d{2}/g);
                    return {
                        soldCount: soldMatch ? soldMatch[1] : null,
                        prices: priceMatch ? priceMatch.slice(0, 5) : [],
                        title: document.title,
                        url: window.location.href,
                    };
                }""")
                print(f"    Sold count: {sold_info['soldCount']}, Prices: {sold_info['prices'][:3]}")
                results.append({
                    "type": "item",
                    "url": item_url,
                    "screenshot": str(screenshot_path),
                    "title": sold_info["title"],
                    "sold_count": sold_info["soldCount"],
                    "prices": sold_info["prices"],
                })

            except Exception as e:
                print(f"    ERROR: {e}")

            item_id += 1
            page.wait_for_timeout(1500)

    browser.close()

manifest = {
    "captured_at": "2026-05-08",
    "total_searches": len(SEARCHES),
    "results": results,
}
manifest_path = OUT_DIR / "manifest.json"
with open(manifest_path, "w") as f:
    json.dump(manifest, f, indent=2)

print(f"\nDone. Manifest: {manifest_path}")
print(f"Screenshots: {len(list(OUT_DIR.glob('*.png')))}")